### PR TITLE
Foreground service player

### PR DESCRIPTION
**Pull request by ouchadam**
- Patches `no-player` to support audio
- Introduces a `player` module; Adds a foreground service to host the player

### CODE REVIEW

File: player/src/main/java/com/example/PlayerService.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            .setSmallIcon(R.drawable.abc_ic_ab_back_material)
+            .build()
+        startForeground(100, notification)
+
+        play("https://dcs.megaphone.fm/HSW5458739817.mp3?key=b05fac12fc24b3ba70beab6e5bef68c3&listener=0193e468-9de6-455c-b033-74f81ee5da94")
+    }
+
+    private fun play(url: String) {
+        val player = PlayerBuilder()
+            .withPriority(PlayerType.EXO_PLAYER)
```
Comment (ouchadam) [inline player/src/main/java/com/example/PlayerService.kt:47]: media player is patched as well but I noticed some strange behaviour around redirects + exoplayer is easier to debug

File: settings.gradle [hunk 2]
```diff
@@ -1 +1 @@
-include ':app', 'no-player'
+include ':app', ':player', 'no-player'
```
Comment (ataulm) [inline settings.gradle:1]: How come copying `no-player` instead of using the lib?

Comment (ouchadam) [reply settings.gradle:1]: we need to patch it to allow audio playback, and by patch I mean comment out a bunch of checks and pass nulls

File: app/build.gradle [hunk 3]
```diff
@@ -30,7 +30,7 @@ android {
 }
 
 dependencies {
-    implementation project(":no-player")
+    implementation project(":player")
```
Comment (ataulm) [inline app/build.gradle:33]: nit single quote

File: player/build.gradle [hunk 4]
```diff
@@ -0,0 +1,17 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+
+android {
+  compileSdkVersion versions.androidSdk.compile
+
+  defaultConfig {
```
Comment (ataulm) [inline player/build.gradle:7]: ~~we can ditch this default config block for library modules~~ maybe this is a lie

File: player/build.gradle [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    minSdkVersion versions.androidSdk.min
+    targetSdkVersion versions.androidSdk.target
+  }
+}
+
+dependencies {
+  implementation project(":no-player")
```
Comment (ataulm) [inline player/build.gradle:14]: nit: single quote

File: player/.gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (ataulm) [inline player/.gitignore:1]: doesn't the root .gitignore take care of this already?

File: player/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (ataulm) [inline player/src/main/AndroidManifest.xml:1]: is the name space necessary?

File: player/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -0,0 +1,2 @@
...
+    package="com.example"/>
```
Comment (ataulm) [inline player/src/main/AndroidManifest.xml:2]: `com.podcats.player`?

File: settings.gradle (hunk 2 above)
Comment (ataulm) [reply settings.gradle:1]: which bit was the audio patch?

Comment (ouchadam) [reply settings.gradle:1]: all the changes to `no-player`, mostly around ignoring the if the `playerView` has been attached

**Pull request merged**